Objects/TI4_Helpers/TI4_SystemHelper.ttslua

```diff
@@ -538,6 +538,18 @@ function systemFromPosition(position)
     end
 end
 
+-- Get systems at positions
+-- @param Map of arbitrary key to {xyz} position
+-- @return Map of same arbitrary key to system attributes or false
+function systemsFromPositions(keysToPositions)
```

Comment:
Instead of casting, we can do the following:
- Iterate through `getAllObjects()` and collect every one with a guid in the systems table.
- Get the position of every one of the above system objects
- Use hexesFromPositions to get the hex ID of each system object
- Use hexesFromPositions to get the hex ID of each input position
- Use the two hex maps to find the system tile for each input position, if any.

This way is all math. Totally justifies throwing a `coroutine.yield(0)` after you call this method, but shouldn't be unstable or need to be a coroutine itself.

What do you think?

Reply:
That seems reasonable, table scans aren't free but a single scan is much, much better than repeated casts.  Agree want to yield after, but this amount of work can be done in a single frame.